src/Adapter/Store/CommandHandler/ToggleStoreStatusHandler.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Store\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\Store\AbstractStoreHandler;
+use PrestaShop\PrestaShop\Core\Domain\Store\Command\ToggleStoreStatusCommand;
+use PrestaShop\PrestaShop\Core\Domain\Store\CommandHandler\ToggleStoreStatusHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Store\Exception\CannotToggleStoreStatusException;
+use PrestaShop\PrestaShop\Core\Domain\Store\Exception\StoreException;
+use PrestaShopException;
+
+/**
+ * Handles command that toggle store status
+ */
+class ToggleStoreStatusHandler extends AbstractStoreHandler implements ToggleStoreStatusHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(ToggleStoreStatusCommand $command): void
+    {
+        $store = $this->getStore($command->getStoreId());
+
+        try {
+            if (false === $store->toggleStatus()) {
```

Comment:
If you accept to use repositories here you can use the `StoreRepository` as well, the repository can implement an `update` and/or a `partialUpdate` method In this case the partial update fits perfectly, the idea is to limit the SQL request and update only the required field (in this case `active`)

Reply:
Using the repository will make the code in the handler cleaner since all the try/catch logic is already handled by the repository itself

Reply:
Here is an example of status toggling with partial update:

https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/Product/CommandHandler/UpdateProductStatusHandler.php#L72-L76

Except in your case the code would be smaller since you don't have to deal with indexation